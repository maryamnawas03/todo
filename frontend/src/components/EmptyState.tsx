// src/components/EmptyState.tsx
import React from "react";
import { Circle } from "lucide-react";

export const EmptyState: React.FC = () => {
  return (
    <div className="text-center py-12">
      <div className="flex justify-center mb-4">
        <Circle className="w-16 h-16 text-gray-300" />
      </div>
      <h3 className="text-xl font-semibold text-gray-700 mb-2">
        No tasks yet
      </h3>
      <p className="text-gray-500">Create your first task to get started</p>
    </div>
  );
};
