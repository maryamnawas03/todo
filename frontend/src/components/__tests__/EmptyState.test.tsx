// src/components/__tests__/EmptyState.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { EmptyState } from '../EmptyState';

describe('EmptyState', () => {
  describe('Rendering', () => {
    it('should render "No tasks yet" heading', () => {
      render(<EmptyState />);
      
      expect(screen.getByText('No tasks yet')).toBeInTheDocument();
    });

    it('should render instructional message', () => {
      render(<EmptyState />);
      
      expect(screen.getByText('Create your first task to get started')).toBeInTheDocument();
    });

    it('should display both heading and description', () => {
      render(<EmptyState />);
      
      expect(screen.getByText('No tasks yet')).toBeInTheDocument();
      expect(screen.getByText(/create your first task/i)).toBeInTheDocument();
    });

    it('should render heading as h3', () => {
      render(<EmptyState />);
      
      const heading = screen.getByRole('heading', { level: 3 });
      expect(heading).toHaveTextContent('No tasks yet');
    });
  });

  describe('Content', () => {
    it('should provide encouraging message to create tasks', () => {
      render(<EmptyState />);
      
      const message = screen.getByText(/create your first task/i);
      expect(message).toBeVisible();
    });

    it('should be displayed when task list is empty', () => {
      render(<EmptyState />);
      
      expect(screen.getByText('No tasks yet')).toBeVisible();
      expect(screen.getByText('Create your first task to get started')).toBeVisible();
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot', () => {
      const { asFragment } = render(<EmptyState />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
