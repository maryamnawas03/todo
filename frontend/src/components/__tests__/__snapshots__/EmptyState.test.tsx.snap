// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Snapshot should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center py-12"
  >
    <div
      class="flex justify-center mb-4"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-circle w-16 h-16 text-gray-300"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
      </svg>
    </div>
    <h3
      class="text-xl font-semibold text-gray-700 mb-2"
    >
      No tasks yet
    </h3>
    <p
      class="text-gray-500"
    >
      Create your first task to get started
    </p>
  </div>
</DocumentFragment>
`;
